src/fragment-generation-utils.js

```diff
@@ -19,13 +19,29 @@ import * as fragments from './text-fragment-utils.js';
 const MAX_EXACT_MATCH_LENGTH = 300;
 const ITERATIONS_BEFORE_ADDING_CONTEXT = 3;
 
+const TIMEOUT_DURATION_MS = 400;  // Desired max run time, in ms
```

Comment:
Should this be parameterized?

Reply:
Good call. Went with a global because it would've been awful to plumb it all the way down into every helper method that checks the timeout.

It might be a good idea later to wrap this all up into a class, but since it's a module with one primary function, I think that's mostly a code style distinction, without much different behavior in practice.

Reply:
It could be parameterized while still being a global var (not constant). The main API would set/reset the global timeout value upon being called.